src/compiler/crystal/tools/doc/generator.cr

```diff
@@ -278,8 +278,13 @@ class Crystal::Doc::Generator
   end
 
   def compute_repository
+    `git rev-parse --is-inside-work-tree >/dev/null 2>&1`
+    unless $?.success?
+      puts "WARNING there is not inside git worktree. some features are not available."
+      return
+    end
+
     remotes = `git remote -v`
```

Comment:
`git remote -v` error message except `fatal: Not a git repository` may be useful for us, should not silence it. So, we should detect whether inside work-tree before `git remote`.